src/Raven.Client/Documents/Queries/Facets/FacetQuery.cs

```diff
@@ -171,17 +31,31 @@ public static FacetQuery Create(string indexName, IndexQueryBase query, string f
 
             return result;
         }
+    }
 
-        private static DynamicJsonValue SerializeFacetsToDynamicJson(IEnumerable<Facet> facets)
-        {
-            var array = new DynamicJsonArray();
-            foreach (var facet in facets)
-                array.Add(facet.ToJson());
+    public abstract class FacetQuery<T> : IndexQueryBase<T>
+    {
+        public string[] FieldsToFetch { get; set; }
 
-            return new DynamicJsonValue
-            {
-                ["Facets"] = array
-            };
-        }
+        public QueryOperator DefaultOperator { get; set; }
+
+        public string DefaultField { get; set; }
+
+        public bool IsDistinct { get; set; }
+
+        /// <summary>
+        /// Index name to run facet query on.
+        /// </summary>
+        public string IndexName { get; set; }
```

Comment:
No, this will be removed, copied it here so I would not have to spend whole day on server-side just to make the code compile.